### PR TITLE
boot: Cleanup

### PR DESCRIPTION
**Pull request by medhefgo**

### CODE REVIEW

File: src/boot/efi/boot.c [hunk 1]
```diff
@@ -104,6 +104,11 @@ enum {
         TIMEOUT_TYPE_MAX    = UINT32_MAX,
 };
 
+enum {
+        IDX_MAX = INT16_MAX,
+        IDX_INVALID,
```
Comment (bluca) [inline src/boot/efi/boot.c:110]: we usually initialize the _INVALID enum to `-EINVAL`

**Review by bluca — APPROVED** [review]

File: src/boot/efi/boot.c (hunk 1 above)
Comment (keszybz) [reply src/boot/efi/boot.c:110]: Yes, please do this so that it can _INVALID can be used in strerror().

Comment (keszybz) [reply src/boot/efi/boot.c:110]: Hmm, so here IDX_INVALID is positive, and the return type is changed to UINTN. Why do this change? It'd seem to make things more confusing.

Comment (medhefgo) [reply src/boot/efi/boot.c:110]: This isn't userspace, so no `strerror()`. And UINTN (== size_t) is used everywhere for array indices and lengths. Using a signed type is more confusing and inconsistent. It also creates unsound signed/unsigned comparisons that require a cast.

If you want to insist, I can set this to UINTN_MAX (which was the effective invalid value before). Ultimately, the value doesn't matter as it's just an internal sentinel.

Comment (keszybz) [reply src/boot/efi/boot.c:110]: OK, the enum here is just to give a name to the constant.

**Pull request merged**